fsspec/spec.py

```diff
@@ -1841,11 +1841,20 @@ def read(self, length=-1):
             length = self.size - self.loc
         if self.closed:
             raise ValueError("I/O operation on closed file.")
-        logger.debug("%s read: %i - %i", self, self.loc, self.loc + length)
         if length == 0:
             # don't even bother calling fetch
             return b""
         out = self.cache._fetch(self.loc, self.loc + length)
+
+        logger.debug(
```

Comment:
This seems like too much information here. Such logging should be controlled by the logger in the cache module, not here, especially since we may have a cacher that does nothing, or no cacher at all for backends that don't support it.

Reply:
Agree, maybe this should be better in `caching.py`

Reply:
Let me backtrack here, I think having this information in one place is more useful for traceability.

```
<File-like object S3FileSystem, file.h5> read: 432000000 - 440000000, total: 276824064, cache hits: 34, cache misses: 33
```

is not that different from 

```
<File-like object S3FileSystem, file.h5> read: 432000000 - 440000000
```

If we move it to the cache implementations we'll have to put those calls and the file objects together (percolate the logs?) in the case we use no cache or the cache does nothing we'll only see zeroes.

Reply:
Since we introspect the state of the cache anyway, maybe we can just have the cache* fields produced by a method on the cache objects, where the BaseCache's implementation returns no numbers at all? Does that sound too complex?

Reply:
That's a good approach, something like this?

```python
logger.debug("%s read: %i - %i, %s,
    self,
    self.loc,
    self.loc + length,
    self.cache.state()
)

```